src/input/reader.rs

```diff
@@ -123,9 +134,7 @@ impl Seek for Reader {
             )),
             Memory(a) => Seek::seek(a, pos),
             Compressed(a) => Seek::seek(a, pos),
-            File(a) => Seek::seek(a, pos),
             Restartable(a) => Seek::seek(a, pos),
-            Vec(a) => Seek::seek(a, pos),
             ExtensionSeek(a) => a.seek(pos),
```

Comment:
You'll need to branch this (and `Reader::is_seekable`) on `MediaSource::is_seekable` for Extension types if you don't want to hit an `unreachable!()` or other panic!